staging/src/k8s.io/mount-utils/mount_helper_common.go

```diff
@@ -82,11 +86,11 @@ func CleanupMountWithForce(mountPath string, mounter MounterForceUnmounter, exte
 // IsNotMountPoint will be called instead of IsLikelyNotMountPoint.
 // IsNotMountPoint is more expensive but properly handles bind mounts within the same fs.
 // if corruptedMnt is true, it means that the mountPath is a corrupted mountpoint, and the mount point check
-// will be skipped
+// will be skipped. The mount point check will also be skipped if the mounter supports it.
 func doCleanupMountPoint(mountPath string, mounter Interface, extensiveMountPointCheck bool, corruptedMnt bool) error {
 	var notMnt bool
 	var err error
-	if !corruptedMnt {
+	if !mounter.canSafelySkipMountPointCheck() && !corruptedMnt {
```

Comment:
```
if mounter.canSafelySkipMountPointCheck() || corruptedMnt {
  // mounter.Unmount
  // removePath if no error from unmount
} else {
  // removePathIfNotMountPoint
  // mounter.Unmount
  // removePathIfNotMountPoint if no error
}
```

Reply:
in previous logic, if unmount returns no error, we still check mountpoint before removing path. Is this really necessary? If umount command returns with no error, does it indicate the path is no longer a mount point? We should just call removePath instead of removePathIfNotMountPoint, right?

Reply:
I think that the previous discussion from [here](https://github.com/kubernetes/kubernetes/pull/109117#issuecomment-1092163588) is related to this question. If we can assume that we can rely on rm command's failure to check if there is something mounted, it would be true.